apps/checkout-app/__tests__/pages/api/webhooks/mollie.test.ts

```diff
@@ -0,0 +1,34 @@
+import handler from "@/checkout-app/pages/api/webhooks/mollie";
+import {
+  mockRequest,
+  setupPollyMiddleware,
+  setupRecording,
+} from "@/checkout-app/test-utils";
+
+describe("/api/webhooks/mollie", () => {
+  const context = setupRecording();
+
+  beforeEach(() => {
+    setupPollyMiddleware(context.polly.server);
+  });
+
+  it("handles invalid requests payments", async () => {
+    const { req, res } = mockRequest("POST");
+
+    await handler(req, res);
+
+    expect(res.statusCode).toBe(400);
+  });
+
+  it("handles requests with invalid orderId", async () => {
+    const { req, res } = mockRequest("POST");
+
+    req.body = {
+      id: "invalid_id",
+    };
+
+    await handler(req, res);
+
+    expect(res.statusCode).toBe(500);
```

Comment:
Not related to test, but I don't think it should be status 500. Invalid ID is probably a bad request (if invalid format) or not found (if not found but proper format)?

Reply:
That's true, the error code should be 400 - the issue is that we catch all errors from Mollie in `verifyPayment`, which could include errors that are not really the fault of the client.

Reply:
Yes, API function validates if the body is correct. For example in previous tests we check if the body contains the `id` and if the API returns 400 if the request didn't have that in the body.

In this case however we can't be sure about the cause of the error, since it's comming from external API. It can be invalid `orderId` (ie. Mollie responds that an order with provided id doesn't exist) but it could be something else (ex. Mollie API is down / timeout / our API keys have expired)

Reply:
Ok, I see.

I think in the future you can also handle some Mollie's errors and pass them down, so if mollie returns 404, you also return 404

Reply:
It's more complicated than that, because Mollie client doesn't expose error codes (I've tested it in debugger) and only gives out error message (that's not declared anywhere isnide the client, so I cannot even compare the messages to check what was the cause of error because the message might change) 🙃

Mollie's build doesn't even export the `ApiError` class they use internally